### ISSUE TITLE
Sign-in/Login - using google  firebase

### ISSUE THREAD

**Issue by krithika2543** [Omkar-Sonawane-23/Tourism#281]
Add a feature that allows users to log in using their Google account through Firebase Authentication. This way, it will help to create more user interaction and the users can now comment and also rate the page and will be also able to interact with other readers.

Set Up Firebase Project:
Create or use an existing project in the Firebase Console.
Enable the Google sign-in method under the Authentication section in the Firebase Console.
Users click the "Login with Google" button.
The frontend handles the Google sign-in process and obtains the ID token, and sends it to the backend.
The backend verifies the ID token and manages the user session.

GSSOC'2024 Contributor

**Comment by prateeksha17** [comment]
Assign this to me, I will fix this issue.

**Comment by krithika2543** [comment]
@Omkar-Sonawane-23 can you please label this as Gssoc24. Actually this step is necessary to create comments section

**Comment by Shubhi-03** [comment]
Can you please label this as Gssoc'24. I want to work on it.

**Comment by sameer-soni** [comment]
firebase is good but most of the people contributing here are new, so they may problems with apis and stuffs and we don't want to reveal those important credentials. So what if we go with basic client side authentication by using localstorage or maybe cookies. What do you guys think?

**Comment by krithika2543** [comment]
Yes that,s a better solution, however now that i think i feel instead of a comments section, i think it would be better to have a FAQ section where any user can ask questions as in a tourism blog webapp this makes more sense. Log/Signin wouldnt have much functionality here. What do you propose

**Comment by krithika2543** [comment]
@sameer-soni Yep it makes sense. I am willing to contribute to this idea

**Comment by sameer-soni** [comment]
@krithika2543 okay, assigning it to you then. Without signing in, users can also access the pages. The reason for the authentication is to have user comments, reviews and their feedbacks of places.

**Comment by Yogesh-ok55** [comment]
I have added this feature in my project many times and i feel i am well experienced to do it ,i have studied your code and i feel i can work to make it look better

**Comment by krithika2543** [comment]
@sameer-soni hey sorry for the delay, i see the frontend part is done. should i now work on the backend

**Comment by priyankMS** [comment]
assign me this issue i can fix them
